pkg/apis/tensorflow/v1/common.go

```diff
@@ -16,8 +16,14 @@ package v1
 
 // SuccessPolicy is the success policy.
 type SuccessPolicy string
+type FailurePolicy string
 
 const (
 	SuccessPolicyDefault    SuccessPolicy = ""
 	SuccessPolicyAllWorkers SuccessPolicy = "AllWorkers"
 )
+
+const (
+	FailurePolicyDefault    FailurePolicy = ""
+	FailurePolicyAllWorkers FailurePolicy = "AllWorkers"
```

Comment:
We can have a FailurePolicyChief = "Chief" and use it by default.

Reply:
I agree, I think we can have following scenarios
1. Based on the Chief: Chief fails, the job fails.
2. Based on the Worker0 (if there is no Chief): Worker0 fails, the job fails.
2. Based on a Worker: If one of the workers fails, the job fails.
3. Based on all Workers: If all of the workers fails, the job fails.

@gaocegege  do you think should we consider Chief in case 2,3 ?
For example, in case3, what if 2 out of 4 workers fail, but Chief also fails? What is the better way to define the semantic?

Reply:
If we can, I think we should.

Reply:
Thanks for the suggestion.
Agree, we should have more failure policy to handle job failure.
I'm going to update it

btw, I've check https://travis-ci.org/github/kubeflow/tf-operator/jobs/699406748 
but still don't know why test failed?

Reply:
FAIL	github.com/kubeflow/tf-operator/pkg/controller.v1/tensorflow	4.601s

Can you pass UT in your local dev machine?

Reply:
You should be able to find the command and reproduce it locally.

Reply:
Thanks for the help. I will try it.